app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,1233 @@
+import employee_app.database;
+import employee_app.email;
+import employee_app.types;
+
+import ballerina/cache;
+import ballerina/http;
+import ballerina/log;
+import ballerina/time;
+
+configurable email:EmailAlertConfig offerEmailConfig = ?;
+
+final cache:Cache userInfoCache = new ('capacity = types:CAPACITY, 'evictionFactor = types:EVICTION_FACTOR);
+final cache:Cache orgStructureCache = new ('capacity = types:CAPACITY, 'evictionFactor = types:EVICTION_FACTOR);
+
+@display {
+    label: "Employee Application",
+    id: "people-ops/employee-application"
+}
+service http:InterceptableService / on new http:Listener(9090) {
+
+    # Request interceptor.
+    # + return - authorization:JwtInterceptor
+    public function createInterceptors() returns http:Interceptor[] {
+        return [new authorization:JwtInterceptor()];
+    }
+
+    # Get basic information of a given active employee.
+    #
+    # + email - Email of the employee
+    # + return - Basic information of the employee or an error
+    resource function get employees/[string email]()
+        returns types:Employee|http:BadRequest|http:InternalServerError|http:NotFound {
+
+        if !email.matches(types:WSO2_EMAIL_PATTERN) {
+            return <http:BadRequest>{
+                body: {
+                    message: string `Input email is not a valid WSO2 email address: ${email}`
+                }
+            };
+        }
+        if userInfoCache.hasKey(email) {
+            any|error cachedEmployee = userInfoCache.get(email);
+            if cachedEmployee is types:Employee {
+                return cachedEmployee;
+            } else if cachedEmployee is cache:Error {
+                log:printError("Error getting employee information", cachedEmployee);
+            }
+        }
+
+        types:Employee|error? employee = database:getEmployee(email);
+        if employee is error {
+            string errorMsg = string `Error getting employee basic information for ${email}!`;
+            log:printError(errorMsg, employee);
+            return <http:InternalServerError>{
+                body: {
+                    message: errorMsg
+                }
+            };
+        }
+        if employee is () {
+            log:printDebug(string `${types:NOT_FOUND_CUSTOM_ERROR}: ${email}`);
+            return <http:NotFound>{
+                body: {
+                    message: string `${types:NOT_FOUND_CUSTOM_ERROR}: ${email}`
+                }
+            };
+        }
+
+        return employee;
+    }
+
+    # Get basic information of employees.
+    #
+    # + filters - Filter objects containing the filter criteria for the query
+    # + 'limit - The maximum number of employees to return
+    # + offset - The number of employees to skip before starting to collect the result set
+    # + return - Basic information of the employees or an error
+    resource function post employees(types:EmployeeFilter? filters, int? 'limit, int? offset)
```

Comment:
```suggestion
    resource function post employees/search(types:EmployeeFilter? filters, int? 'limit, int? offset)
```

provides clarity to what the resource method actually does

Reply:
@rdulmina this method gives all the details about the employee. But we need to filter them for specific purposes in other services. Since the filter amount is large we've decided to go with `POST` than `GET` with query params.

Reply:
Yes resource method `POST` is fine. I was saying `post employees/search`provide more clarity to what this method actually intend to than `post employees`. `post employees` is feels like add new employee resource

Reply:
+1 but with the requirement, for now we've internally agreed to go with this.

Reply:
What is the reason to go with `post employees` instead of `post employees/search`. Also it is not straight forward to change the resource path after we publish it. It's better to change now

Reply:
We had a discussion today for this again and agreed to go with `employees/search`. 
The initial reason to go with that was to have an endpoint `/employees` with `/employess/[email]`.